processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -285,31 +311,53 @@ void add(@Nullable final Filter filter)
             }
           }
         }
+        preFilters.add(filter);
+
+      }
+    }
 
-        if (requiredColumns.contains(outputColumnName)) {
-          // Rewrite filter post-unnest if possible.
+    final FilterSplitter filterSplitter = new FilterSplitter();
+
+    // non-unnest case
+    // OR CASE
+    List<Filter> preFilterList = new ArrayList<>();
+    List<Filter> postFilterList = new ArrayList<>();
+    if (queryFilter instanceof OrFilter) {
+      for (Filter filter : ((OrFilter) queryFilter).getFilters()) {
+        if (filter.getRequiredColumns().contains(outputColumnName)) {
           final Filter newFilter = rewriteFilterOnUnnestColumnIfPossible(filter, inputColumn, inputColumnCapabilites);
           if (newFilter != null) {
-            // Add the rewritten filter pre-unnest, so we get the benefit of any indexes, and so we avoid unnesting
-            // any rows that do not match this filter at all.
-            preFilters.add(newFilter);
+            preFilterList.add(newFilter);
+            postFilterList.add(newFilter);
+          } else {
+            postFilterList.add(filter);
```

Comment:
If it's not re-writable, then I don't think the OR can be pushed down at all.  If you only pushdown the things that don't refer to the output of the unnest, you run the risk of not including some rows that should've been included.  It should be relatively easy to test for this.  If there's a query with an `MV_TO_STRING(unnested_dim, 'a') = 'b' OR other_dim = 'a'`, I don't think it'll believe that it can remap it.  If it pushes down the `other_dim = a` without attempting to push down the first part of the OR, it will not include a row where `nested_dim` is `["b"]` and `other_dim` is `"c"`.

Reply:
Added test cases to mimic all the different cases